docs/dev/getting_started.md

```diff
@@ -0,0 +1,141 @@
+# Getting Started
+
+This document explains how to get started with developing for Kubernetes Ingress.
+It includes how to build, test, and release ingress controllers.
+
+## Dependencies
+
+The build uses dependencies in the `ingress/vendor` directory, which
+must be installed before building a binary/image. Occasionally, you
+might need to update the dependencies. 
+
+This guide requires you to install the[godep](https://github.com/tools/godep)dependency
+tool.
+
+Check the version of `godep` you are using and make sure it is up to date.
+```console
+$ godep version
+godep v74 (linux/amd64/go1.6.1)
+```
+
+If you have an older version of `godep`, you can update it as follows:
+```console
+$ cd $GOPATH/src/ingress
+$ go get github.com/tools/godep
+$ cd $GOPATH/src/github.com/tools/godep
+$ go build -o godep *.go
+```
+
+This will automatically save the dependencies to the `vendor/` directory.
+```console
+$ cd $GOPATH/src/ingress
+$ godep save ./...
+```
+
+In general, you can follow [this guide](https://github.com/kubernetes/kubernetes/blob/release-1.5/docs/devel/godep.md#using-godep-to-manage-dependencies)to update dependencies.
+To update a particular dependency, eg: Kubernetes:
+```console
+$ cd $GOPATH/src/k8s.io/ingress
+$ godep restore
+$ go get -u k8s.io/kubernetes
+$ cd $GOPATH/src/k8s.io/kubernetes
+$ godep restore
+$ cd $GOPATH/src/k8s.io/kubernetes/ingress
+$ rm -rf Godeps
+$ godep save ./...
+$ git [add/remove] as needed
+$ git commit
+```
+
+## Building
+
+All ingress controllers are built through a Makefile. Depending on your
+requirements you can build a raw server binary, a local container image,
+or push an image to a remote repository.
+
+In order to use your local Docker, you may need to set the following environment variables:
+* export TAG=0.0 # or whatever you want the version to be named
+* export DOCKER=docker
+* export REGISTRY=index.docker.io
```

Comment:
The default is set to `gcr.io/google_containers` which can cause a problem for first time developers or anyone not using gcloud. I can set it to something more like <your docker registry>.